src/components/header/renderer.tsx

```diff
@@ -741,7 +746,22 @@ class Header extends React.Component<Props, State> {
           </PortalWithState>
         </section>
         <section className="right-section">
-          <PortalWithState closeOnEsc defaultOpen={this.props.showExample}>
+          <PortalWithState
+            closeOnEsc
+            defaultOpen={this.props.showExample}
+            onOpen={() => {
+              const node = ReactDOM.findDOMNode(this.examplePortal.current);
+              node.scrollTop = this.props.lastPosition;
+              node.addEventListener('scroll', () => {
```

Comment:
Why do we need this if we set the state when we close the modal?

Reply:
Okay, let me explain you what's happening.

So, when the user opens the portal we are getting the div element in the DOM which shows example as a node.
Then we are scrolling the div to the position which is stored in the redux store.
The above lines are responsible for accessing the DOM and making it scroll to the stored position, the default position is set to 0 in the default state of the redux. So if the user opens the portal for the first time then he stays at the top of the portal.

The setScrollPosition() function which is called onClose of the portal is responsible for storing the location in the redux store.

And when the user reopens the modal, it again fetches the location from the store and scroll to that particular position where the user was there before.

What happens when the user toggles between the modes is, we again access the DOM using the react reference and scroll it to the top.